app/fetch_preconfig.js

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import RNFetchBlob from 'react-native-fetch-blob';
+
+import {Client4} from 'mattermost-redux/client';
+import {General} from 'mattermost-redux/constants';
+import EventEmitter from 'mattermost-redux/utils/event_emitter';
+
+const HEADER_X_VERSION_ID = 'X-Version-Id';
+const HEADER_X_CLUSTER_ID = 'X-Cluster-Id';
+const HEADER_TOKEN = 'Token';
+
+Client4.doFetchWithResponse = async (url, options) => {
```

Comment:
Instead of duplicating all this, we could add a method to Client4 in the redux library to allow you to override the call to fetch like we do when the redux library needs to know about system emojis or call localization code, but this code is stable so it's fine to duplicate for now